pkg/controllers/cloud/cloudprovideraccount_controller_test.go

```diff
@@ -148,5 +150,57 @@ var _ = Describe("CloudProviderAccount Controller", func() {
 			err = reconciler.processDelete(&testAccountNamespacedName)
 			Expect(err).Should(HaveOccurred())
 		})
+		It("Secret watcher", func() {
+			var err error
+			ch := make(chan error)
+			reconciler.clientset = fakewatch.NewSimpleClientset()
+			credential := `{"accessKeyId": "keyId","accessKeySecret": "Secret"}`
+			err = os.Setenv("POD_NAMESPACE", testSecretNamespacedName.Namespace)
+			Expect(err).ShouldNot(HaveOccurred())
+			// Create secret
+			// Created a secret using clientset because watcher is created using clientset. So inorder to know
+			// watcher if the secret is modified secret should be created using clientset. Secret should be created using
```

Comment:
Does this convey message correctly? 
```Create a secret using both fakeClient and clientSet, since secret watcher uses cleintset while reconciler using client from controller run-time. Both secret watcher and reconciler should be able to retrieve the secret object.```

Reply:
Solved.